### DISCUSSION TITLE
anyone maintaining code for docker containers?

### DISCUSSION THREAD

**Discussion by itaru2622** [bluesky-social/atproto#2334] [General Discussion]
I'm using latest codes  to get running bluesky on containers in my env.

I found that features can work with dev-env as expected but not with container.
through investigations, I found  gaps, such as belows, between code of dev-env and those of containers.

by filling these gaps one by one,  the containers got starting to work little by little, but not all features yet.
I guess it need to find all the gaps and feed them back into the container codes to get the full features working on the containers.

---
found gaps regarding atproto/services/bsky/*.js
- it lacks data-plane, bsync, RepoSubscription in atproto/services/bsky/*.js  but dev-env has those:
  https://github.com/bluesky-social/atproto/blob/f45eef3414f8827ba3a6958a7040c7e38bfd6282/packages/dev-env/src/bsky.ts#L43-L51
  https://github.com/bluesky-social/atproto/blob/f45eef3414f8827ba3a6958a7040c7e38bfd6282/packages/dev-env/src/bsky.ts#L88-L93
  
- RepoSubscription uses BackgroundQueue but it is not exported then BackgroundQueue cannot be involved from container.  
  in dev-env.  it is imported as below, with source accessing style.
  https://github.com/bluesky-social/atproto/blob/f45eef3414f8827ba3a6958a7040c7e38bfd6282/packages/dev-env/src/bsky.ts#L9
  
found gaps on timing of creating DB table( by migrations/*):
- dev-env creates tables earlier than container code, then containers get failure to use DB, but dev-env success.
  earlier creactions are done not just for bsky but also ozone and others.
  https://github.com/bluesky-social/atproto/blob/f45eef3414f8827ba3a6958a7040c7e38bfd6282/packages/dev-env/src/bsky.ts#L75-L83

**Comment by bnewbold** [comment]
The short answer is that running the `main` branch code directly isn't really supported. We have the dev-env setup for development, and we do "distribution releases" of services like the PDS and Ozone via separate repositories (https://github.com/bluesky-social/pds and https://github.com/bluesky-social/ozone).

The majority of code is in this repository, but supporting folks running the latest version of `main` at any point of time would be a lot of additional maintenance and support work. You are welcome to try and make it work, but we will probably focus on periodic releases of the "distribution" versions for now.

Another thing to note is that we pretty frequently have large feature branches in this repository, so even `main` might not be fully up to date.

**Reply by itaru2622** [reply]
@bnewbold thank you for your response. I understood  situation.
at this moment, it needs to deal with individually.
when you got resource, please maintain codes for docker container as part of offical release at timing of assigning  versions. since docker is major platform to run software.

**Reply by itaru2622** [reply]
this post aims to record what it needs for bsky working on docker.

it needs to patch https://github.com/bluesky-social/atproto/blob/main/services/bsky/api.js to involve DB, bsync, dataplane, reposubscription, like dev-env had them.

my patch is located below:
https://github.com/itaru2622/bluesky-selfhost-env/blob/master/patching/105-atproto-services-for-docker.diff

**Comment by itaru2622** [comment]
closing this topic, by getting current situation described in the above bnewbold's comment.
